comp/dogstatsd/server/component.go

```diff
@@ -0,0 +1,67 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package server implements a component to run the dogstatsd server
+package server
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/dogstatsd"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"go.uber.org/fx"
+)
+
+// team: Agent Metrics Logs
+
+// Component is the component type.
+type Component interface {
+
+	// Start starts the dogstatsd server
+	Start(demultiplexer aggregator.Demultiplexer) error
+	// Stop stops the dogstatsd server
+	Stop()
+
+	// Capture starts a new dogstatsd traffic capture
+	Capture(p string, d time.Duration, compressed bool) error
+
+	// IsCaputreOngoing returns true if a capture is ongoing
+	IsCaputreOngoing() bool
+
+	// GetCapturePath returns the capture destination path
+	GetCapturePath() (string, error)
+
+	// GetJSONDebugStats returns a json representation of debug stats
+	GetJSONDebugStats() ([]byte, error)
+
+	// GetJSONDebugStats gets the DsdServerDebug instance which provides metric stats
+	GetDebug() *dogstatsd.DsdServerDebug
+
+	// EnableMetricsStats enables metric stats tracking
+	EnableMetricsStats()
+
+	// DisableMetricsStats disables metric stats tracking
+	DisableMetricsStats()
```

Comment:
As these 3 methods work together, it would be nice to have a struct `MetricStats`.

Reply:
I agree. I think this is worth re-visting once the `server` is componentized. 

Perhaps a good solution is to make the dogstatsd debug it's own sub-componenet injected into the server and the runtime settings (once that is also a componenet). This will allow us to remove these functions from the server's public interface and only expose it where needed.